### ISSUE TITLE
createDataProcessor

### ISSUE THREAD

**Issue by WojakGra** [DHTMLX/scheduler#62]
```javascript
scheduler.createDataProcessor({
        event: {
            create: function (data) {
                console.log(data);
                return new scheduler.Promise(function (resolve, reject) {
                    $.post(
                        "api.php",
                        { action: "test", data: data },
                        function (result) {
                            return resolve({
                                action: result.action,
                                tid: result.tid,
                                data: result.task,
                            });
                        },
                        "json"
                    );
                });
            },
            update: function (data, id) {
                console.log(data);
                console.log(id);
                return new scheduler.Promise(function (resolve, reject) {
                    $.post(
                        "api.php",
                        { action: "test", data: data },
                        function (result) {
                            return resolve({ action: result.action });
                        },
                        "json"
                    );
                });
            },
            delete: function (id) {
                console.log(id);
                return new scheduler.Promise(function (resolve, reject) {
                    $.post(
                        "api.php",
                        { action: "test", data: id },
                        function (result) {
                            return resolve({ action: result.action });
                        },
                        "json"
                    );
                });
            },
        },
        mode: "JSON",
    });
```
Upper code not working as intentended because is shows a console error
```
ajax.js:219          POST http://localhost/calendar/undefined 404 (Not Found)
```

**Comment by AlexKlimenkov** [comment]
Hi @WojakGra ,

I've confirmed the issue, we'll fix it in the upcoming bugfix update.

Meanwhile, please try initializing the dataprocessor with a function, like this:

```
scheduler.createDataProcessor(function(entity, action, data){
	switch(action) {
		case "create":
			console.log(data);
			return new scheduler.Promise(function (resolve, reject) {
				$.post(
					"api.php",
					{ action: "test", data: data },
					function (result) {
						return resolve({
							action: result.action,
							tid: result.tid,
							data: result.task,
						});
					},
					"json"
				);
			});
		break;
		case "update":
			console.log(data);
			console.log(id);
			return new scheduler.Promise(function (resolve, reject) {
				$.post(
					"api.php",
					{ action: "test", data: data },
					function (result) {
						return resolve({ action: result.action });
					},
					"json"
				);
			});
		break;
		case "delete":
			console.log(id);
			return new scheduler.Promise(function (resolve, reject) {
				$.post(
					"api.php",
					{ action: "test", data: id },
					function (result) {
						return resolve({ action: result.action });
					},
					"json"
				);
			});
		break;
	}
});
```

Then it should work correctly.
